homeassistant/components/twitch/sensor.py

```diff
@@ -56,40 +63,56 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Twitch platform."""
-    channels = config[CONF_CHANNELS]
-    client_id = config[CONF_CLIENT_ID]
-    client_secret = config[CONF_CLIENT_SECRET]
-    oauth_token = config.get(CONF_TOKEN)
-
-    try:
-        client = await Twitch(
-            app_id=client_id,
-            app_secret=client_secret,
-            target_app_auth_scope=OAUTH_SCOPES,
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2024.2.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize entries."""
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
         )
-        client.auto_refresh_auth = False
-    except TwitchAuthorizationException:
-        LOGGER.error("Invalid client ID or client secret")
-        return
+    )
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
 
-    if oauth_token:
-        try:
-            await client.set_user_authentication(
-                token=oauth_token, scope=OAUTH_SCOPES, validate=True
-            )
-        except MissingScopeException:
-            LOGGER.error("OAuth token is missing required scope")
-            return
-        except InvalidTokenException:
-            LOGGER.error("OAuth token is invalid")
-            return
+    await session.async_ensure_token_valid()
+
+    app_id = implementation.__dict__[CONF_CLIENT_ID]
+    app_secret = implementation.__dict__[CONF_CLIENT_SECRET]
```

Comment:
Same comments are the config flow on accessing the ID and leaving out the secret.